src/racer/cargo.rs

```diff
@@ -639,9 +639,22 @@ pub fn get_crate_file(kratename: &str, from_path: &Path) -> Option<PathBuf> {
 
     if let Some(tomlfile) = find_cargo_tomlfile(from_path.to_path_buf()) {
         // look in the lockfile first, if there is one
+        // check if there's a workspace first.
         trace!("get_crate_file tomlfile is {:?}", tomlfile);
+        let mut workspace: Option<String> = None;
+        if let Some(toml) = parse_toml_file(&tomlfile) {
```

Comment:
Oh, you might need to save off the parsed toml file to avoid borrowck issues:

```rust
let toml = parse_toml_file(&tomlfile);
let workspace = toml.and_then( // .. as before
```